src/BaseDrawableMesh.ts

```diff
@@ -0,0 +1,109 @@
+import { Euler, Group, Material, Mesh, Vector3 } from "three";
+import { IDrawableObject } from "./types";
+
+/**
+ * Abstract base class for drawable 3D mesh objects.
+ *
+ * Provides default implementations for some methods in the IDrawableObject
+ * interface, including handling for visibility, transformations, and cleanup.
+ *
+ * As a default, subclasses should call `this.addChildMesh(mesh)` to register
+ * any meshes that should be managed by the base class.
+ */
+export default abstract class BaseDrawableMesh implements IDrawableObject {
+  /**
+   * Pivot group that all child meshes are parented to. Transformations are
+   * applied to this group.
+   */
+  protected readonly meshPivot: Group;
+  protected scale: Vector3;
+  protected flipAxes: Vector3;
+
+  /**
+   * Child meshes that are part of this drawable object.
+   */
+  protected readonly meshes: Set<Mesh>;
```

Comment:
this may be a strange question but why are they managed separately if they are in meshPivot.children?

Reply:
Premature optimization, haha. `meshPivot.children` is an array and I thought it would be faster for some operations (checking if an object exists) to use a Set. I did do some checks and `meshPivot.add` is safe against multiple object adds, so it's likely that Three already has a set representation internally.